app/uk/gov/hmrc/agentregistrationfrontend/config/AppConfig.scala

```diff
@@ -1,10 +1,78 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package uk.gov.hmrc.agentregistrationfrontend.config
 
 import javax.inject.{Inject, Singleton}
 import play.api.Configuration
+import play.api.mvc.{Call, Request}
+import sttp.model.Uri
+import sttp.model.Uri.UriContext
+import uk.gov.hmrc.http.StringContextOps
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+
+import scala.concurrent.duration.Duration.Infinite
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Try
 
 @Singleton
-class AppConfig @Inject()(config: Configuration) {
-  val welshLanguageSupportEnabled: Boolean = config.getOptional[Boolean]("features.welsh-language-support").getOrElse(false)
+class AppConfig @Inject()(servicesConfig: ServicesConfig, configuration: Configuration) {
+
+  val thisFrontendBaseUrl: String = ConfigHelper.readConfigAsValidUrlString("urls.this-frontend", configuration)
+  val feedbackFrontendBaseUrl: String = ConfigHelper.readConfigAsValidUrlString("urls.bas-frontend-sign-in", configuration)
+  val basFrontendSignBaseInBaseUrl: String = ConfigHelper.readConfigAsValidUrlString("urls.bas-frontend-sign-in", configuration)
+  val basFrontendSignOutUrlBase: String = ConfigHelper.readConfigAsValidUrlString("urls.bas-frontend-sign-out", configuration)
+
+  def signInUri(continueUri: Uri): Uri = {
+    val queryParams = Map(
+      "continue" -> continueUri,
```

Comment:
for sign in uris, the parameter `continue` is deprecated by [bas-gateway-frontend](https://github.com/hmrc/bas-gateway-frontend?tab=readme-ov-file#get-sign-in), the parameter `continue_url` is **required**, can we change the name of this param to `continue_url`?